src/engine/Service.cpp

```diff
@@ -384,3 +362,44 @@ ProtoResult Service::makeNeutralElementResultForSilentFail() const {
   }
   return {std::move(idTable), resultSortedOn(), LocalVocab{}};
 }
+
+// ____________________________________________________________________________
+void Service::verifyVariables(
+    const nlohmann::json& j,
+    const ad_utility::LazyJsonParser::Generator& gen) const {
+  if (!j["head"].contains("vars") || !j["head"]["vars"].is_array()) {
+    throwErrorWithContext("JSON result does not have the expected structure",
+                          gen.details().first100_);
```

Comment:
You could also include the complete `"head"` clause  in this error message, because that's what causes the problem here, so something like
`... does not have the expected structere, as its "head" section is not according to the SPARQL standard. The "head"section is" .... dump the subjson here`.